packages/clay-core/src/picker/Picker.tsx

```diff
@@ -382,6 +382,22 @@ export function Picker<T extends Record<string, any> | string | number>({
 		);
 	}
 
+	const getComponentWidth = () => {
+		let menuMinWidth = '160px';
+		let menuWidth = 'auto';
+
+		if (typeof width === 'number') {
+			menuMinWidth = `${triggerRef.current?.clientWidth || 0}px`;
+			menuWidth = `${width}px`;
+		} else if ((triggerRef.current?.clientWidth || 0) > 160) {
+			menuMinWidth = `${triggerRef.current?.clientWidth}px`;
+		}
+
+		return {menuMinWidth, menuWidth};
+	};
+
+	const {menuMinWidth, menuWidth} = getComponentWidth();
```

Comment:
We don't need to declare this as a function and call it. We can just set the variables directly in the render function.

Reply:
Oh! I got it! 
I'll send the fix in a minute